src/lib/device-online-state.ts

```diff
@@ -181,6 +245,10 @@ class DeviceOnlineStateManager {
 					req: root,
 				},
 			})
+			.tap(() => {
+				const endAt = Date.now();
+				this.emit('change', { startAt, endAt, uuid, newState });
```

Comment:
We should also emit on failure, but with the extra info that it failed (so we can track failure rate alongside success rate)

Reply:
👍